scripts/zones/Upper_Jeuno/npcs/Mapitoto.lua

```diff
@@ -31,15 +32,15 @@ function onTrade(player,npc,trade)
 end
 
 function onTrigger(player,npc)
+    local fsaQuest = player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.FULL_SPEED_AHEAD) 
     local fullSpeedAheadStatus = player:getCharVar("[QUEST]FullSpeedAhead")
-    local hasTrainersWhistle = player:hasKeyItem(tpz.ki.TRAINERS_WHISTLE)
 
-    if hasTrainersWhistle then
+    if fsaQuest == QUEST_COMPLETED then
         player:startEvent(10226)
-    elseif fullSpeedAheadStatus == 2 then -- Retry
-        player:startEvent(10224, 1)
     elseif fullSpeedAheadStatus == 3 then -- Complete
-        player:startEvent(10225, tpz.ki.TRAINERS_WHISTLE)
+        player:startEvent(10225, tpz.ki.TRAINERS_WHISTLE, 15533, ID.npc.MAPITOTO)
```

Comment:
Could be wrong, but I think you bumped up your quest vars, so talking to Mapitoto after trying once will result in completion. 😉